components/builder-http-gateway/src/http/middleware.rs

```diff
@@ -138,30 +138,49 @@ impl Authenticated {
         self
     }
 
-    fn authenticate(&self, req: &mut Request, token: SessionToken) -> IronResult<Session> {
-        let mut request = SessionGet::new();
-        request.set_token(token);
-        let conn = req.extensions.get_mut::<XRouteClient>().unwrap();
-        match conn.route::<SessionGet, Session>(&request) {
-            Ok(session) => {
-                self.validate_session(&session)?;
-                Ok(session)
+    fn authenticate(&self, req: &mut Request, token: &str) -> IronResult<Session> {
+        // Test hook - always create a valid session
+        if env::var_os("HAB_FUNC_TEST").is_some() {
+            debug!(
+                "HAB_FUNC_TEST: {:?}; calling session_create_short_circuit",
+                env::var_os("HAB_FUNC_TEST")
+            );
+            return session_create_short_circuit(req, token);
+        };
+
+        // Check for a valid personal access token
+        if bldr_core::access_token::is_access_token(token) {
```

Comment:
Is there any possibility that the base64-encoding of an internal sessionsrv token could start with `TOKN`? Perhaps we should make `ACCESS_TOKEN_PREFIX` include some non-base64 character like `!`.

Reply:
No, there's no possibility of that happening. The possibility of that happening is the same as having an encrypted version of a plaintext reveal several consecutive characters of the plaintext in the encrypted output, which is nil. The bigger issue is needing the prefix at all - it's a bit of a ugly hack - ideally I'd like to eliminate it completely, but there are several tradeoffs to doing so at the moment.

Reply:
Update - this explanation isn't quite right.  However the issue is moot because there is a significant advantage of using a non-base-64 char in the prefix - ie, we can eliminate all the other prefix chars, since we only need to differentiate between access token and session token (and github token, for now).  So we'll save 3 extra characters from the token size!  Which brings us back to the size before the duration change was made.. pretty cool :)